File: arrays/minimum_index_sum_of_two_lists.rb
# @param {String[]} list1
# @param {String[]} list2
# @return {String[]}
def find_restaurant(list1, list2)

    count_hash = Hash.new(0)

    list1.each do |str|
        if list2.include?(str)
            count_hash[str] += list1.index(str)
            count_hash[str] += list2.index(str)
        end
    end

    count_hash.select { |k, v| v == count_hash.values.min }.keys
end

File: arrays/minimum_index_sum_of_two_lists.py
class Solution:
    def findRestaurant(self, list1: List[str], list2: List[str]) -> List[str]:
        from collections import defaultdict
        count_dict = defaultdict(int)

        for str in list1:
            if str in list2:
                count_dict[str] = list1.index(str) + list2.index(str)

        min_sum = min(count_dict.values()) if count_dict else 0

        return [k for k, v in count_dict.items() if v == min_sum]